### PR TITLE
Challenge50: dotnet

### LINKED ISSUE #1243: Have a .net dll based RE challenge!
### Context

DLLs and exes can be hard to debug. What about hidden secrets over there?

Can we have a challenge whiich is cross-compiled using .Net runtime and a few DLLs to have a secret inside similar like the golang/rust challenges?

- [x] Add .net based binary to the [wrongsecrets/binaries ](https://github.com/OWASP/wrongsecrets-binaries)repowith crosscompiling for the various OSes
- [x]  Add a challenge here that uses the binary (See contributing.md and the code of the other binary challenges).

### PR DESCRIPTION
**Pull request by commjoen**
### What kind of changes does this PR include?

-   [ ] Fixes or refactors
-   [x] A new challenge
-   [ ] Additional documentation
-   [ ] Something else

#### Description

This PR adds challenge50

### Relations

Closes #1243

### Checklist:

-   [x] All the contributions made are solely the work of me and my co-authors
-   [x] I tested the changes in this PR (if applicable)
-   [x] I added unit tests to ensure my change works (when change in Java or on front-end code)
-   [ ] I added UI tests to ensure my UI changes work (when change in the overall UI, not needed if just adding a challenge)
-   [x] The PR passes pre-commit hooks and automated tests

### CODE REVIEW

File: .github/scripts/docker-create.sh [hunk 1]
```diff
@@ -298,6 +298,17 @@ generate_test_data() {
   fi
 }
 
+download_dot_net_binaries() {
+  echo "downloading dotnet binaries"
+  rm ../../src/main/resources/executables/wrongsecrets-dotne*
+  curl -L -o ../../src/main/resources/executables/wrongsecrets-dotnet https://github.com/OWASP/wrongsecrets-binaries/releases/download/0.1.0/wrongsecrets-dotnet
```
Comment (bendehaan) [inline .github/scripts/docker-create.sh:304]:
```suggestion
  curl -L -o ../../src/main/resources/executables/wrongsecrets-dotnet https://github.com/OWASP/wrongsecrets-binaries/releases/download/$BINARY_VERSION/wrongsecrets-dotnet
```

**Review by bendehaan — COMMENTED** [review]
LGTM, maybe make the Version in the script a parameter?

File: .github/scripts/docker-create.sh (hunk 1 above)
Comment (commjoen) [reply .github/scripts/docker-create.sh:304]: Where do you want to maintain the variable? Till now I am not sure if we will have a another release soon ;-)

Comment (commjoen) [reply .github/scripts/docker-create.sh:304]: Feel free to add,can you set the version on the top of the script as well please :-)?

**Pull request merged**